schema.sql

```diff
@@ -1,11 +1,50 @@
 CREATE EXTENSION IF NOT EXISTS "uuid-ossp";
 
+CREATE TABLE courses (
+  course_id serial PRIMARY KEY,
+  course_code varchar(5) NOT NULL,
+  course_name text
+);
+
+CREATE TABLE tracks (
+  track_id serial PRIMARY KEY,
+  track_code varchar(2) NOT NULL,
+  track_name text
+);
+
+CREATE TABLE timezones (
+  tzone_id serial PRIMARY KEY,
+  tzone_code varchar(5) NOT NULL,          -- e.g EST, CET, MIT,
+  tzone_name text,                         -- e.g Eastern Standard Time, Central European Time, Marquesas Islands Time
+  gmtoffset numeric(4,2) NOT NULL          -- e.g -5, 1, -9.50
+);
+
+
 CREATE TABLE users (
-  pkey UUID NOT NULL DEFAULT uuid_generate_v1(),
-  firstname text NOT NULL CHECK (LENGTH(TRIM(FROM firstname)) > 0),
+  user_id serial, -- change datatype from serial to uuid DEFAULT uuid_generate_v4 ()
+  firstname text NOT NULL CHECK (LENGTH(TRIM(BOTH ' ' FROM firstname)) > 0), -- should we use the checks in the form instead of here?
   lastname text,
```

Comment:
Not a big deal at all, but I think we should use varchar() for some of these shorter fields since: 
* varchar only has an overhead of a single byte vs two for text
* varchar can be indexed; text can't 

There's some more detail in this post: https://stackoverflow.com/questions/25300821/difference-between-varchar-and-text-in-mysql

Let me know what you think. Also happy to go with text since the benefits are really only relevant for an application with a lot of data.

Reply:
I agree with you. Perhaps the only field where we should go with text as data type is `about me`.

Reply:
👍  sounds good